docs/app/Examples/modules/Transition/Types/index.js

```diff
@@ -6,23 +6,30 @@ import ExampleSection from 'docs/app/Components/ComponentDoc/ExampleSection'
 
 const TransitionTypesExamples = () => (
   <ExampleSection title='Types'>
+
     <ComponentExample
       title='Transition'
       description='A Transition animates a single child by toggling the the visible prop.'
       examplePath='modules/Transition/Types/TransitionExampleTransition'
     >
       <Message warning>
-        <p>Do not unmount a Transition child or else it cannot be animated.</p>
+        <p>An animated component must not unmount unexpectedly or the transition will not run as intended.</p>
         <Message.List>
           <Message.Item>
-            Use the <code>unmountOnHide</code> prop to unmount the child after the animation exits.
+            If you need to animate children as they mount and unmount, use a <code>Transition.Group</code>. 
+          </Message.Item>
+          <Message.Item>
+            It is paramount that each child of a <code>Transition.Group</code>
+            has an unique <code>key</code> prop that stays constant through rerenders.
+            In particular, it is likely that <code>keys</code> based on pseudo-random identifiers or simple loop increment variables will lead to unexpected behavior.
```

Comment:
I think a link to the React docs is good but I would still be a bit more verbose about the keys, especially since some of the devs adopting Semantic UI may not be ReactJS experts. 

In my case I was using proper React keys in the sense described in the docs: a simple loop variable that was unique among siblings. It was just that when my list was modified and the keys were reset, Transition.Group lost track of which components to animate.